packages/html/src/server/getHTMLFromFragment.ts

```diff
@@ -0,0 +1,46 @@
+import type { Node, Schema } from '@tiptap/pm/model'
+import { DOMSerializer } from '@tiptap/pm/model'
+
+/**
+ * Returns the HTML string representation of a given document node.
+ *
+ * @param doc - The document node to serialize.
+ * @param schema - The Prosemirror schema to use for serialization.
+ * @returns A promise containing the HTML string representation of the document fragment.
+ *
+ * @example
+ * ```typescript
+ * const html = getHTMLFromFragment(doc, schema)
+ * ```
+ */
+export async function getHTMLFromFragment(
+  doc: Node,
+  schema: Schema,
+  options?: { document?: Document },
+): Promise<string> {
+  try {
+    const { Window } = await import('happy-dom')
+
+    if (options?.document) {
+      // The caller is relying on their own document implementation. Use this
+      // instead of the default happy-dom-without-node library.
```

Comment:
If the provided DOM implementation gives errors, should we commit to fix them? Or should we let developers know that we only officially support happy-dom?

Reply:
Actually will remove this and add JSDocs mentioning happy-dom.